taichi/runtime/llvm/llvm_context.cpp

```diff
@@ -817,6 +860,71 @@ void TaichiLLVMContext::update_runtime_jit_module(
     }
   }
 
+  if (arch_ == Arch::amdgpu) {
+    for (auto &f : *module) {
+      bool is_kernel = false;
+      const std::string func_name = f.getName().str();
+      if (starts_with(func_name, "runtime_")) {
+        mark_function_as_amdgpu_kernel(&f);
+        is_kernel = true;
+      }
+      if (!is_kernel && !f.isDeclaration())
+        f.setLinkage(llvm::Function::PrivateLinkage);
+    }
+    std::vector<llvm::Function *> global_func;
```

Comment:
I dont quite find a definition for `global function` in LLVM, there's `external function` only. 

Judged by `llvm::CallingConv::AMDGPU_KERNEL`, do you mean AMD GPU's device function?

Reply:
1. Ohhhh
Yeah, you're right.  `llvm::CallingConv::AMDGPU_KERNEL` has the similar function to `getOrInsertNamedMetadata("nvvm.annotations")` which is used in CUDA.
2. `kernel_function` may be a better choice.